nixos/modules/services/misc/ollama.nix

```diff
@@ -3,12 +3,19 @@ let
   inherit (lib) types;
 
   cfg = config.services.ollama;
-  ollamaPackage = cfg.package.override {
+
+  stableOverrides = {
     inherit (cfg) acceleration;
     linuxPackages = config.boot.kernelPackages // {
       nvidia_x11 = config.hardware.nvidia.package;
     };
   };
+  unstableOverrides = {
+    inherit (cfg) acceleration;
+  };
+
+  ollamaPackage = cfg.package.override (prev:
+    if prev ? linuxPackages then stableOverrides else unstableOverrides);
```

Comment:
This is clever. Maybe too clever: I didn't know `.override` can be used with a function, and spent 20' now thinking about scopes and reading `customisation.nix` 🙃. I also struggle to argue whether this is an implementation detail or a part of the public interface

Reply:
> spent 20' now thinking about scopes and reading `customisation.nix`

So did you not realize that `prev ? linuxPackages` was referring to the previous package arguments? If so, it seems the solution is to change the name to make it more obvious.

> I also struggle to argue whether this is an implementation detail or a part of the public interface

I'm not sure I understand what you mean. Are you asking if passing a function to `override` is intended behavior or an implementation detail? I found it documented [here](https://ryantm.github.io/nixpkgs/using/overrides/), which seems to imply that it's intended to be used as such.

Reply:
> So did you not realize that prev ? linuxPackages was referring to the previous package arguments? If so, it seems the solution is to change the name to make it more obvious.

No, in the current version that is quite clear

> I found it documented [here](https://ryantm.github.io/nixpkgs/using/overrides/), which seems to imply that it's intended to be used as such.

👍🏻